aptos-move/framework/src/natives/aggregator_natives/aggregator.rs

```diff
@@ -0,0 +1,217 @@
+    context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert!(args.len() == 2);
+
+    // Get aggregator fields and a value to add.
+    let value = pop_arg!(args, u128);
+    let aggregator_ref = pop_arg!(args, StructRef);
+    let (handle, key, limit) = get_aggregator_fields(&aggregator_ref)?;
+    let id = AggregatorID::new(handle, key);
+
+    // Get aggregator.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+    let aggregator = aggregator_data.get_aggregator(id, limit);
+
+    aggregator.add(value)?;
+
+    // NOTE(Gas): O(1) cost: simple addition.
+    let cost = gas_params.base_cost;
+
+    Ok(NativeResult::ok(cost, smallvec![]))
+}
+
+pub fn make_native_add(gas_params: AddGasParameters) -> NativeFunction {
+    Arc::new(move |context, ty_args, args| native_add(&gas_params, context, ty_args, args))
+}
+
+/***************************************************************************************************
+ * native fun read(aggregator: &Aggregator): u128;
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct ReadGasParameters {
+    pub base_cost: InternalGas,
+}
+
+fn native_read(
+    gas_params: &ReadGasParameters,
+    context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert!(args.len() == 1);
+    let aggregator_ref = pop_arg!(args, StructRef);
+
+    // Extract fields from aggregator struct reference.
+    let (handle, key, limit) = get_aggregator_fields(&aggregator_ref)?;
+    let id = AggregatorID::new(handle, key);
+
+    // Get aggregator.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+    let aggregator = aggregator_data.get_aggregator(id, limit);
+
+    let value = aggregator.read_and_materialize(aggregator_context.resolver, &id)?;
+
+    // NOTE(Gas): O(1) cost: serialization/deserialization and potential
+    // resolving to storage.
+    let cost = gas_params.base_cost;
+
+    Ok(NativeResult::ok(cost, smallvec![Value::u128(value)]))
+}
+
+pub fn make_native_read(gas_params: ReadGasParameters) -> NativeFunction {
+    Arc::new(move |context, ty_args, args| native_read(&gas_params, context, ty_args, args))
+}
+
+/***************************************************************************************************
+ * native fun sub(aggregator: &mut Aggregator, value: u128);
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct SubGasParameters {
+    pub base_cost: InternalGas,
```

Comment:
Recently I removed the "_cost" postfix from all parameter names. Could you please also remove them from yours? Much appreciated!